consoleApp/src/main/java/com/infoshare/database/FileDb.java

```diff
@@ -0,0 +1,112 @@
+package com.infoshare.database;
+
+import com.infoshare.domain.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+
+public class FileDb implements DB {
+    private static final String VOLUNTEER_DB_FILE_NAME = "Volunteer.csv";
+    private static final String REQUEST_DB_FILE = "NeedRequest.csv";
+    private static final SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss"); // uzywamy df do formatowania i parsowania dat
+
+    public FileDb() {
+    }
+
+    @Override // zapis / aktualizacja danych  wolontariusza
+    public void saveVolunteer(Volunteer volunteer) throws IOException {
+        // Sprawdzamy czy wolontariusz istnieje
+
+        if (getVolunteer(volunteer.getEmail()) == null) {
```

Comment:
Ten warunek wygląda na to, że nie działa. Porównanie maila można by zrobić wyciągając aktualną listę wolontariuszy i sprawdzać, czy na liście jest taki mail (jeśli założymy, że unikalność będziemy zapewniać po mailu).

Reply:
metoda getVolunteer robi to wlasnie to co mowisz , wyciaga liste volontariuszy z bazy i zwraca wolontariusza jezeli istnieje albo null jesli nie.